src/Pages/Loading.js

```diff
@@ -0,0 +1,109 @@
+/* global BigInt */
+
+import TotalVAR from './TotalVAR';
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from "react-router-dom";
+import { createClient } from "urql"
+
+let protocolInfo = []
+
+export let allProtocols = []
+export let totalVaR = 0;
+
+const APIURL = "https://api.thegraph.com/subgraphs/name/greg-torrington/greg-v3"
+const assetQuery = `
+query {
+  assets {
+    id
+    name
+  	count
+    users
+  }
+}
+`
+
+const client = createClient({
+    url: APIURL
+})
+
+var navigate;
+
+function Loading(){
+
+    let usersQuery
+
+    async function fetchBlockChainData() {
```

Comment:
This function records the subgraph data and stores it in an array. It is structured so that protocolInfo will have 3 values, name of the specific protocol, the contracts to that specific protocol in an array, and the total aaVaR of the protocol. Inside the contracts array will be each contract, its name, a another array holding all the users inside that contract, and the aaVaR of that contract. After the protocolInfo array is populated, it will be pushed to allProtocols which holds every protocol for the application.